analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ParameterValidationInYieldShouldBeWrapped.CSharp10.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Tests.Diagnostics
+{
+    public static class InvalidCases
+    {
+        public static IEnumerable<string> Foo(string something) // FN
+        {
+            ArgumentNullException.ThrowIfNull(something); // FN sec
+
+            yield return something;
+        }
+
+        public static IEnumerable<int> YieldBreak(int a) // FN
+        {
+            if (a < 0)
+            {
+                ArgumentNullException.ThrowIfNull(a); // FN sec
```

Comment:
This test case is misleading. In your example, an `ArgumentException` should be raised.

Copy the test case `Foo` from above. This is just about `yield return` vs. `yield break`.